src/unique_ptr.rs

```diff
@@ -9,61 +9,119 @@ use core::marker::PhantomData;
 use core::mem::{self, MaybeUninit};
 use core::ops::{Deref, DerefMut};
 use core::pin::Pin;
+use cxx::type_id;
 
-/// Binding to C++ `std::unique_ptr<T, std::default_delete<T>>`.
+/// Rust representation of the default deleter of std::unique_ptr, `std::default_delete<T>`
+#[derive(Default, Copy, Clone, Debug)]
 #[repr(C)]
-pub struct UniquePtr<T>
-where
-    T: UniquePtrTarget,
-{
+pub struct DefaultDeleter([u8; 0]);
+
+unsafe impl ExternType for DefaultDeleter {
+    type Id = type_id!("std::default_delete");
+    type Kind = Trivial;
+}
+
+#[cfg(not(target_os = "macos"))]
+#[repr(C)]
+struct UniquePtrInner<T, D> {
+    deleter: D,
     repr: MaybeUninit<*mut c_void>,
     ty: PhantomData<T>,
 }
 
-impl<T> UniquePtr<T>
+#[cfg(target_os = "macos")]
+#[repr(C)]
+struct UniquePtrInner<T, D> {
+    repr: MaybeUninit<*mut c_void>,
+    deleter: D,
+    ty: PhantomData<T>,
+}
```

Comment:
This is not something I am willing to assume about the internal layout of std::unique_ptr.

Can you suggest some other way to implement this feature?